### DISCUSSION TITLE
5 questions about captions and training

### DISCUSSION THREAD

**Discussion by abetteridea** [bmaltais/kohya_ss#1204] [Q&A]
**1.** Is there a difference between training for 40 repeats over 1 epoch and 1 repeat over 40 epochs?

**2.** What is the math behind regularisation images Repeats? Why is there 1 repeat by default for regularisation images and 40 repeats for training? Are we suggested to have 40 regularisation images for each training image?

**3.** kohya-ss says [here](https://github.com/kohya-ss/sd-scripts/issues/85#issuecomment-1399145611)

> As you might know, if you use the caption files, the 'activation word' in the folder name is ignored (such as sks frog in the folder name 10_sks frog, or a character name etc.).

Is this true for this GUI?

**4.** Is the reg images folder name also ignored when caption files are present?

**5.** On a side note, do we need caption files for regularisation images? Does it make sense to generate regularisation images with the same captions as training images (without the instance prompt/token) or only use a class prompt when generating them?

**Comment by abetteridea** [comment]
Found an answer to my **questions 1 & 2** from kohya-ss [here](https://github.com/kohya-ss/sd-scripts/discussions/182#discussioncomment-4946693)

> There are two purpose for repeats.
> 
> - Balance the number of training images to the regularization images. The number of regularization images is larger than the training, so it is required to repeat training images for using all regularization images in the epoch.
> - Control 'weight' over folders. If you have high quality images and low quality images, you can set higher number of repeats for high quality images, and lower for low quality.

**Questions 3-5** remain a mystery!

**Reply by belladoreai** [reply]
Did you find an answer to these questions? I've been wondering about the exact same things!

I'll venture some guesses according to my current understanding:

3. Yes
4. Yes
5. Yes

**Comment by psklf** [comment]
>  Is the reg images folder name also ignored when caption files are present?

Yes.
.





>  On a side note, do we need caption files for regularisation images? Does it make sense to generate regularisation images with the same captions as training images (without the instance prompt/token) or only use a class prompt when generating them?

I have tried reg imgs without caption and got good result.
